test/unit/test_logger.py

```diff
@@ -126,3 +126,28 @@ def test_cyan_text():
     x = '{}{}{}'.format(colorama.Fore.CYAN, 'foo', colorama.Style.RESET_ALL)
 
     assert x == logger.cyan_text('foo')
+
+
+def test_markup_detection_pycolors_0(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '0')
+    assert not logger.should_do_markup()
+
+
+def test_markup_detection_pycolors_1(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '1')
+    assert logger.should_do_markup()
+
+
+def test_markup_detection_is_tty(monkeypatch, mocker):
+    monkeypatch.delenv('PY_COLORS', raising=False)
+    monkeypatch.setenv('TERM', 'xterm')
+    stdout_mock = mocker.patch("sys.stdout")
```

Comment:
```suggestion
    monkeypatch.setattr('sys.stdout.isatty', lambda: True)
```

Reply:
Oh, even better. This suggests using a CM: https://docs.pytest.org/en/latest/monkeypatch.html

Reply:
@webknjaz I had a rough experience with pytest CM so I would prefer if introducing its use in molecule test would be done by someone else, outside the scope of this patch. So far there is no existing code in molecule that uses pytest context manager.

PS. I am not againt CM, just that I already wasted too much time trying to make it work (and ended up with "good" old style).

Reply:
Just a few weeks ago I faced pytest-mock breaking test suite in an exceptionally weird manner in Ansible Core tests. Those patched objects were leaking into pytest internals badly.

And it's still weird to combine several patching tools in one place.

If you want to keep mocker, please ensure to always unpatch everything upon leaving the test function:
```python
    mocker.resetall()
    mocker.stopall()
```
Ref: https://github.com/ansible/ansible/pull/52372/files#diff-7d19c1624d0977d32233184491b27e1bR156

Reply:
Thanks for the hint! I am sure this may prove handly in the future. Meanwhile I already removed one of the mocks so we should be safe.